libyul/backends/evm/SSAControlFlowGraphBuilder.cpp

```diff
@@ -567,7 +569,15 @@ std::vector<SSACFG::ValueId> SSAControlFlowGraphBuilder::visitFunctionCall(Funct
 			SSACFG::Operation result{{}, SSACFG::BuiltinCall{_call.debugData, builtin, _call}, {}};
 			for (auto&& [idx, arg]: _call.arguments | ranges::views::enumerate | ranges::views::reverse)
 				if (!builtin.literalArgument(idx).has_value())
-					result.inputs.emplace_back(std::visit(*this, arg));
+				{
+					if (!m_literalsAsPushConstants && std::holds_alternative<Literal>(arg))
+					{
+						auto const lit = m_graph.newLiteral(debugDataOf(currentBlock()), std::get<Literal>(arg).value.value());
```

Comment:
```suggestion
						auto const literal = m_graph.newLiteral(debugDataOf(currentBlock()), std::get<Literal>(arg).value.value());
```

Hum...I assume the creation of the literal in the ssa graph is needed here because it has no `ValueID` at this point, right? I wonder if there is a better place to do that,  but probably not.

Reply:
That was my reasoning ;)